### PR TITLE
wxGUI/animation: fix export output animation file to avi format

### PR DESCRIPTION
**Pull request by tmszi**
Fix bug reported in the issue #835.

### CODE REVIEW

**Comment by petrasovaa** [comment]
Could you explain how this fixes the issue?

**Comment by tmszi** [comment]
> Could you explain how this fixes the issue?

When I tried export animation as avi file `g.gui.animation` frame completly freeze (without crash or return error). `writeAvi()` call ffmpeg from `subprocess.Popen` which block GUI. Good idea I'll add it.

**Comment by tmszi** [comment]
> Could you explain how this fixes the issue? If it is running in the background, maybe there should be a way to inform user export is happening (wx.BusyInfo, or BusyCursor), otherwise users may be confused.

I solved it using `wx.CallAfter()`.

File: gui/wxpython/animation/controller.py [hunk 1]
```diff
@@ -621,10 +622,17 @@ def _export(self, exportInfo, decorations):
 
         # export
         pilImages = [WxImageToPil(image) for image in images]
-        busy = wx.BusyInfo(_("Exporting animation, please wait..."),
+        aelf.busy = wx.BusyInfo(_("Exporting animation, please wait..."),
```
Comment (nilason) [inline gui/wxpython/animation/controller.py:625]: typo

Comment (tmszi) [reply gui/wxpython/animation/controller.py]: Thanks. I've already fixed it.

File: lib/python/imaging/images2avi.py [hunk 2]
```diff
@@ -113,15 +117,28 @@ def writeAvi(filename, images, duration=0.1, encoding='mpeg4',
...
 
     if S.wait():
-        # An error occurred, show
-        print(outPut)
-        print(S.stderr.read())
         # Clean up
         _cleanDir(tempDir)
-        raise RuntimeError("Could not write avi.")
+        if bg_task:
+            return gscript.decode(outPut) + '\n' + gscript.decode(
```
Comment (petrasovaa) [inline lib/python/imaging/images2avi.py:123]: It seems gscript is not defined.

Comment (tmszi) [reply lib/python/imaging/images2avi.py:125]: Thanks. I've already fixed it.

**Comment by tmszi** [comment]
> Apart from undefined gscript it works. I wonder if this works on Windows (given ffmpeg is on path), but I can't test now.

This error doesn't occur on OS Windows. I tested (without patch and with patch) it on OS Windows and it worked properly without errors and crashes.

**Comment by petrasovaa** [comment]
> > Apart from undefined gscript it works. I wonder if this works on Windows (given ffmpeg is on path), but I can't test now.
> 
> This error doesn't occur on OS Windows. I tested (without patch and with patch) it on OS Windows and it worked properly without errors and crashes.

Thanks for testing!

**Comment by tmszi** [comment]
I tested it again @petrasovaa, but my patch doesn't fix this bug (master branch). It's possible that I ran the `g.gui.animation` tool in a different way (from cmd or console) during my testing, and the bug seemed to be resolved. Because the error only occurs if the `g.gui.animation` tool is run from the wxGUI Modules tab tree. My mistake I apologize for that. 

The solution is the remove this line from code: [`wx.GetApp().Yield()`](https://github.com/OSGeo/grass/blob/master/gui/wxpython/animation/controller.py#L625). Could you roll it back before this my change?  But it wok correctly in both cases (as bg thread or without thread) but with removed this line `wx.GetApp().Yield()` from code. I tested it on GNU/Linux, MS Windows OS. Could you confirm this error and test the suggested change? Thanks.

**Comment by petrasovaa** [comment]
I am not able to reproduce the problem, no matter how I launch the tool. I would keep the changes there and I agree removing that line would make sense, Yield() is always problematic. @veroandreo can you reproduce the problem with current master? Please try launching animation tool in different ways.

**Comment by veroandreo** [comment]
I don't get any error when exporting to .avi format, neither calling the animation tool from gui nor from cli. I'm using current master (freshly compiled).

**Comment by tmszi** [comment]
> I don't get any error when exporting to .avi format, neither calling the animation tool from gui nor from cli. I'm using current master (freshly compiled).

Thanks for your time. The bug still persists during my testing. Could @petrasovaa @veroandreo test it again by following the procedure below? Thanks.

**To Reproduce:**

Tested with **slovakia3d_grass7** location (export raster maps as animation), PERMANENT mapset.

There is still a specific condition when this bug occur:

1. Fresh start GRASS GIS with wxGUI `grass79 --gui` (this is an **important step**)
2. On the Layer Manager go to Modules page (tab) and launch Animation tool from modules tree
3. Try export animation to avi format

**Note:**

If you start GRASS GIS according step 1 and then close wxGUI and lauch wxGUI again from virtual terminal, `g.gui` command and repeat step 2., animation export work correctly. Or if you start GRASS GIS in the text mode `grass79 --text` and then launch wxGUI `g.gui` and repeat step 2., animation export work correctly too.


**System description:**
```
GRASS version: 7.9.dev                                                          
Code revision: f9399a7e5                                                        
Build date: 2020-08-28                                                          
Build platform: x86_64-pc-linux-gnu  
Python: 3.6.11                                                                  
wxPython: 4.0.7.post2 
```

**Comment by veroandreo** [comment]
I tested as you indicate in the steps to reproduce and GRASS crashes. This is what I get in the terminal:
```
Stopped                 grass79dev --gui
```

**Pull request merged**